### ISSUE TITLE
Multimer predicting a homomer

### ISSUE THREAD

**Issue by agustin-ormazabal** [aqlaboratory/openfold#453]
Hello there! I am trying to perform a multimer prediction using precomputed MSAs by employing this script:

```
#SBATCH --job-name=OpenFold_%A
#SBATCH --time=03:00:00
#SBATCH -e logs_apd/err.txt
#SBATCH -o logs_apd/out.txt
#SBATCH --gpus=a100
#SBATCH -N 1
#SBATCH --ntasks=8
#SBATCH --mem=128000

export TRITON_CACHE_DIR=/tmp/triton_autotune_cache
export CUTLASS_PATH=/nfs/research/agb/research/agustin/2024/Mycoplasma_pneumoniae_proteome/OpenFold/All_positives/cutlass/
export LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/nfs/research/agb/research/agustin/software/localcolabfold/colabfold-conda/lib/python3.10/site-packages/deepspeed/ops/deepspeed4science

#source /homes/${USER}/.bashrc
module load cuda/11.8.0
module load cudnn/8.6.0.163-11.8
module load cudnn-8.0.4.30-11.1-gcc-9.3.0-bbr3kjv
module load ant-1.10.0-gcc-9.3.0-xzxbcc6
module load gcc/11.2.0

# This variables are not going to change in every cicle

data="/hps/nobackup/agb/research/francesco/data/alphafold/20240326_databases"
openfold="/nfs/research/agb/research/agustin/2024/Mycoplasma_pneumoniae_proteome/OpenFold/openfold"
output_dir="output"
fasta_dir="/nfs/research/agb/research/agustin/2024/Mycoplasma_pneumoniae_proteome/OpenFold/All_positives"
msas="/nfs/research/agb/research/agustin/2024/Mycoplasma_pneumoniae_proteome/OpenFold/All_positives/msas"

python3 ${openfold}/run_pretrained_openfold.py \
    $fasta_dir \
    $data/pdb_mmcif/mmcif_files/ \
    --uniref90_database_path $data/uniref90/uniref90.fasta \
    --mgnify_database_path $data/mgnify/mgy_clusters_2022_05.fa \
    --pdb_seqres_database_path $data/pdb_seqres/pdb_seqres.txt \
    --uniref30_database_path $data/uniref30/UniRef30_2021_03 \
    --uniprot_database_path $data/uniprot/uniprot.fasta \
    --bfd_database_path $data/bfd/bfd_metaclust_clu_complete_id30_c90_final_seq.sorted_opt \
    --use_precomputed_alignments $msas \
    --config_preset "model_1_multimer_v3" \
    --model_device "cuda:0" \
    --output_dir $output_dir \
    --long_sequence_inference
```

As an input, I am using the fasta file provided as an example in the Openfold tutorial, but modifying it in order to have:

```
>6KWC
GSTIQPGTGYNNGYFYSYWNDGHGGVTYTNGPGGQFSVNWSNSGEFVGGKGWQPGTKNKVINFSGSYNPNGNSYLSVYGWSRNPLIEYYIVENFGTYNPSTGATKLGEVTSDGSVYDIYRTQRVNQPSIIGTATFYQYWSVRRNHRSSGSVNTANHFNAWAQQGLTLGTMDYQIVAVQGYFSSGSASITVS
>6KWC
GSTIQPGTGYNNGYFYSYWNDGHGGVTYTNGPGGQFSVNWSNSGEFVGGKGWQPGTKNKVINFSGSYNPNGNSYLSVYGWSRNPLIEYYIVENFGTYNPSTGATKLGEVTSDGSVYDIYRTQRVNQPSIIGTATFYQYWSVRRNHRSSGSVNTANHFNAWAQQGLTLGTMDYQIVAVQGYFSSGSASITVS
>6KWC
GSTIQPGTGYNNGYFYSYWNDGHGGVTYTNGPGGQFSVNWSNSGEFVGGKGWQPGTKNKVINFSGSYNPNGNSYLSVYGWSRNPLIEYYIVENFGTYNPSTGATKLGEVTSDGSVYDIYRTQRVNQPSIIGTATFYQYWSVRRNHRSSGSVNTANHFNAWAQQGLTLGTMDYQIVAVQGYFSSGSASITVS
>6KWC
GSTIQPGTGYNNGYFYSYWNDGHGGVTYTNGPGGQFSVNWSNSGEFVGGKGWQPGTKNKVINFSGSYNPNGNSYLSVYGWSRNPLIEYYIVENFGTYNPSTGATKLGEVTSDGSVYDIYRTQRVNQPSIIGTATFYQYWSVRRNHRSSGSVNTANHFNAWAQQGLTLGTMDYQIVAVQGYFSSGSASITVS
```

 Consequently, I am expecting a tetramer, but instead I obtained a monomer.
 I am also employing the alignment files provided in the tutorial.

**Comment by C-de-Furina** [comment]
With my experience, you should change chain names in fasta files to distinguish them, even they are the same chains. Like:
```
>6KWC_A
ASDFG
>6KWC_B
ASDFG
>6KWC_C
ASDFG
>6KWC_D
ASDFG
```

I do not know why but it works.  You also should have four MSA folders.

**Comment by agustin-ormazabal** [comment]
> With my experience, you should change chain names in fasta files to distinguish them, even they are the same chains. Like:
> 
> ```
> >6KWC_A
> ASDFG
> >6KWC_B
> ASDFG
> >6KWC_C
> ASDFG
> >6KWC_D
> ASDFG
> ```
> 
> I do not know why but it works. You also should have four MSA folders.

@C-de-Furina Thank you very much for your answer. I noticed the same a few days ago and forgot to comment it here. Indeed, this makes the trick (both having four different names for every chain and having one folder of MSAs for each of them, even when they all contain the same files). I strongly recommend adding this clarification to the documentation, since that input format is different regarding other implementations such as AF2, AlphaPulldown, or Unifold.

I also noticed that, in case you are using pre-computed MSAs, the folders corresponding to every chain must be placed in a subfolder called "alignments" inside your output path. Moreover, the uniprot.sto file must be named "uniprot_hits.sto" to be recognised (otherwise, you'll receive a quite cryptic error, difficult to associate with the filename). For instance, AlphaPulldown output for that file is named "uniprot.sto", so it is necessary to rename it to make things work.

I hope this helps to others.
Agus.
